studio/src/editor-panel.js

```diff
@@ -94,20 +92,12 @@ export default class EditorPanel extends LitElement {
         return true;
     }
 
-    discardChanges(refresh = true) {
+    async discardChanges() {
         if (!this.hasChanges) return;
         this.fragmentStore.discardChanges();
-        this.hasChanges = false;
-        if (refresh) this.refresh();
-    }
-
-    aemAction(action, reset = false) {
-        return async function () {
-            this.disabled = true;
-            const ok = await action();
-            if (ok && reset) this.hasChanges = false;
-            this.disabled = false;
-        };
+        Store.fragments.discard.set(true);
+        await this.updateComplete;
```

Comment:
this is causing render nothing / then render everything forcing fields to refresh their values from the fragment.